sympy/core/expr.py

```diff
@@ -989,7 +989,8 @@ def _eval_endpoint(left):
             return B - A
 
         value = B - A
-
+        if type(a) == int:
+            a=S(a)
```

Comment:
This change should not be here. The calling code should pass in a sympified value.

Reply:
made the change in calling code